pkg/syncer/resource_syncer.go

```diff
@@ -257,7 +257,11 @@ func NewResourceSyncer(config *ResourceSyncerConfig) (Interface, error) {
 		UpdateFunc: syncer.onUpdate,
 		DeleteFunc: syncer.onDelete,
 	}, func(obj interface{}) (interface{}, error) {
-		resourceUtil.MustToMeta(obj).SetManagedFields(nil)
+		objMeta, err := meta.Accessor(obj)
```

Comment:
On the face of it, this changes the behaviour if the given object doesn’t have the expected metadata fields — the previous version would panic, the new version ignores the error. This is unlikely anyway, and if there is no metadata then there is nothing to update so ignoring the error is fine, but it would be worth adding a comment to mention that errors aren’t returned on purpose.